lib/srv/alpnproxy/local_proxy.go

```diff
@@ -37,6 +38,7 @@ import (
 // LocalProxy allows upgrading incoming connection to TLS where custom TLS values are set SNI ALPN and
 // updated connection is forwarded to remote ALPN SNI teleport proxy service.
 type LocalProxy struct {
+	*sync.Mutex
```

Comment:
Is this intentional? Feels dangerous to expose `Lock/Unlock` to the outside world.